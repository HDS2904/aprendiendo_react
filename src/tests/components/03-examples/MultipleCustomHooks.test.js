import { prettyDOM, render } from "@testing-library/react";
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks";
import { useCounter } from "../../../hooks/useCounter";
import { useFetch } from "../../../hooks/useFetch";

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas en <MultipleCustomHooks/>', () => {
	
	// beforeEach contiene lo que se ejecutara antes de cada test
	beforeEach( () => {
		useCounter.mockReturnValue({
			state: 10,
			increment: () => {}
		})
	})

	test('Debe mostrarse correctamente', () => {

		useFetch.mockReturnValue({
			data: null,
			loading: true,
			error: null
		})
		
		const wrapper = render( <MultipleCustomHooks /> );

		expect( wrapper.container ).toMatchSnapshot()

		// const etiqueta = wrapper.container.querySelector( 'button' );
		// console.log( prettyDOM( etiqueta ) ); // mostrar la etiqueta extraida de 

	})

	test('Debe de mostrar la informaciÃ³n', async () => {

		useFetch.mockReturnValue({
			data: [{
				author: 'Jonathan',
				quote: 'Programando un mundo'
			}],
			loading: false,
			error: null
		})

		const wrapper = render( <MultipleCustomHooks /> );
		const items = wrapper.container.getElementsByClassName( 'alert' ) // extrae un array de los items con clase 'alert'
		expect( items.length ).toBe( 0 );

		// wrapper.getByText('Programando un mundo') // Forma mas sencilla de evaluar que exista una etiqueta con el contenido
		expect( wrapper.container.querySelector( 'p' ).textContent ).toBe( 'Programando un mundo' ) // selecciona la equiqueta p y trae su contenido para luego evaluar con el expect.tobe
		expect( wrapper.container.querySelector( 'footer' ).textContent ).toBe( 'Jonathan' );
	})
	
})


// Nota:
// existe un problema con el render de @testing-library/react, pues no toma una foto
// de los onclick="" de las etiquetas como lo hace el shallow de enzyme