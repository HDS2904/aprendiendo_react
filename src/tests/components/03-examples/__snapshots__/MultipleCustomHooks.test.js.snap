// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks/> Debe mostrarse correctamente 1`] = `
<div>
  <div>
    <h1>
      BreakingBad Quotes
    </h1>
    <hr />
    <div
      class="alert alert-info text-center"
    >
      Loading...
    </div>
    <button
      class="btn btn-primary"
    >
      Siguiente
    </button>
  </div>
</div>
`;
